### PR TITLE
feat: backend analytics chart

### PR DESCRIPTION
**Pull request by ChielokaCode**
> ## How should this be manually tested?
> **Try accessing a protected route**
> 
> * without being authenticated.
> * with an expired bearer token
> * with an invalid token
>   **The expected results is to get a 401 Unauthenticated response**
> 
> ## Checklist of what you did
> * [x]  Check that user is authenticated before accessing below endpoints
> * [x]  Created endpoint to get summary data of total users, total revenue, new users, active users
> * [x]  Created endpoint to get line chart data of months and the totalSales for each month
> * [x]  Created endpoint to get Bar chart data of months and the totalSales for each month
> * [x]  Created endpoint to get Pie chart data of months, totalSales for each month and the totalSales converted to degrees of each month
>  **The expected result is to get a 200 Successful response
>  **The expected error is to get a 404 Not Found error response
> 
> ## Reference Issue
> [[FEAT]: Implement Backend Support for Analytics Charts on Dashboard](https://github.com/hngprojects/hng_boilerplate_java_web/issues/39)

### CODE REVIEW

**Comment by ChielokaCode** [comment]
@Am0du Please review the PR?

**Comment by ChielokaCode** [comment]
@billmal071

**Review by Fifetoyi — APPROVED** [review]
lgtm

File: src/main/resources/db/migration/V7__sale_database.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE sale (
+    id BIGINT AUTO_INCREMENT PRIMARY KEY,
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V7__sale_database.sql:2]: update the version of your migration file. V7 already exist

**Pull request closed without merging**